### ISSUE TITLE
bicycle=dismount vs bicycle=yes

### ISSUE THREAD

**Issue by limes11** [cyclosm/cyclosm-cartocss-style#393]
Hello all, 

New here, so apologies if this has already been discussed elsewhere. On hw=pedestrian, the default bicycle access value in many countries is dismount. Adding bicycle=yes (meaning cycling at modest pace) does not change the rendering of the road. I don't think that's ideal. Besides the risk cycling in pedestrian areas poses to pedestrians, local police also has an eye on that and likes to dish out fines. I would be very happy if the map told me whether I had to get off the bike or not. 
(Obviously that's a country-specific issue. But that's the case for other things, too.)

Best, John

**Comment by Phyks** [comment]
Hi!

Thanks for the report! The default is indeed that `highway=pedestrian` is a `dismount` for bicycles (see https://wiki.openstreetmap.org/wiki/OSM_tags_for_routing/Access_restrictions#Default). Taking into account country-specific default restrictions is near to impossible in a general purpose map such as CyclOSM.

This is true however that we don't have any special render for `highway=pedestrian` with a `bicycle=yes` tag. 

Example is https://www.openstreetmap.org/way/54967586.

@Florimondable maybe we could keep the large grey border but use a walking speed color?

For the record, relevant style code is https://github.com/cyclosm/cyclosm-cartocss-style/blob/master/roads.mss#L2037-L2079.

Best

**Comment by Florimondable** [comment]
I agree it would be nice to be rendered.
@Phyks walking speed and pedestrian colors are the same.
(there is an optimization that could be done in the lines you show btw).

**Comment by Phyks** [comment]
I'm bad with colors, sorry :/ I was thinking about the green we have here https://www.cyclosm.org/#map=20/48.86674/2.34451/cyclosm (no motor vehicle?).

**Comment by Florimondable** [comment]
Yes it's no motor vehicle. You don't see the difference ?

**Comment by Phyks** [comment]
I had a very bad screen at hand, and the combination of my color perception, small color areas in the legend and the screen made me confusing it :) On a nice screen now there is no doubt !

**Comment by limes11** [comment]
Apologies for the delay. I am a bit confused about these default access values for all countries in the wiki (https://wiki.openstreetmap.org/wiki/OSM_tags_for_routing/Access_restrictions#Default). There, the value is “no” rather than “dismount”. However, “bicycle=no” clearly turns a hw=pedestrian on cyclosm noncyclable, so “hw=pedestrian” and “hw=pedestrian+bicycle=no” are rendered differently. I also checked it in Denmark where the default is bicycle=no on hw=pedestrian, and the rendering is still the same, independently of whether Danish mappers add a bicycle=dismount or not. So, it seems that cyclosm does not actually work with these access values. Or, are they wrong in the wiki?

The problem of bicycle=dismount vs. bicycle=no goes deeper. As explained in the wiki, many mappers use “no” but what they actually mean is “dismount”. I highly doubt that it is forbidden to push a bicycle in a Danish pedestrian area. Some routers therefore also route over ways with a bicycle=no tag simply presuming that it should be fine to walk the bike. 

So, the actual distinction which would be nice to be shown on the map is between “dismount” and “yes”. In this context: Some hw=pedestrian that I have recently worked on carry a bicycle:conditional=yes @ (some time period) tag. It is great that this is actually spelled out on the map. However, it looks a bit silly as the way is marked as always cyclable anyways.

About the colours: I have a moderate red-green color blindness and therefore similar problems distinguishing colours. If I focus on this, I of course see the difference, but it takes me a much bigger effort than it does for "healthy" people.

**Comment by Florimondable** [comment]
The difference between `bicycle=no` and `dismount` is most of the time the board stating it.
If the board order to `dismount` use dismount if it forbid bicycle use `no`.

You're right we have an issue with `bicycle=dismount not` rendered, it should be the same as `bicycle=no`.
example https://www.openstreetmap.org/way/431657731

**Comment by limes11** [comment]
I disagree about this. I know the situation well in the UK and in Germany. In both countries, the usual "bicycle prohibited" sign, that is, the round white one with a bicycle in the center and a red circle, means that cycling is forbidden but it is allowed to push. (Whether using your bike as a  scooter is allowed, is a different matter.) In fact, in these countries, explicit dismount signs are rare and it is unclear if they are even legal. I am sure it is different in some other countries, but I would assume it is similar in most. The same applies to the round white one with the red circle forbidding entry for all vehicles. Cycling forbidden, pushing is fine.

I think the only good way to deal with this is by differentiating between yes, dismount and no.
But using dismount on hw=pedestrian which is intuitively most logical contradicts the wiki....

**Comment by Florimondable** [comment]
The same in France, but we tag what's on the ground not in the law book. We tag the traffic sign not there interpretation.

**Comment by limes11** [comment]
Understand. In that case, I agree with what you said earlier, dismount should be rendered as no which also means that all hw=pedestrian are rendered as noncyclable. It is then up to the user to understand this as "Push your bike if that's allowed in your country".

**Comment by Phyks** [comment]
I would not focus on the `bicycle=no` case, as this can have very different meaning in reality (you can push your bike on a `highway=pedestrian` with `bicycle=no` but you cannot do anything on a `highway=primary` with `bicycle=no`). Handling all of these would probably clutter the map. Same stands for `bicycle=dismount` as in the case presented here, they are often used in an interchangeable way.

`highway=pedestrian` default access restriction for bikes is (equivalent to) `dismount`. Therefore, a pedestrian (with its own rendering) is not expected to be bikable.

What about, on the contrary, rendering `bicycle=yes` tag? I would expect an `highway=pedestrian` + `bicycle=yes` to be basically the same quality as a cycle street (except that there will likely be tons of pedestrians, but this is already indicated by the `pedestrian` rendering). So an `highway=pedestrian` with `bicycle=yes` could be rendered with blue arrows, same as cycle street.

**Comment by limes11** [comment]
"Therefore, a pedestrian (with its own rendering) is not expected to be bikable."

That is exactly the point. hw=pedestrian should have the same rendering as hw=pedestrian+bicycle=no (or the rare hw=pedestrian+bicycle=dismount) but it does not. Instead, hw=pedestrian has the same rendering as hw=pedestrian+bicycle=yes, which is not good.

"as this can have very different meaning in reality (you can push your bike on a highway=pedestrian with bicycle=no but you cannot do anything on a highway=primary with bicycle=no)"

This is true, but it could be managed easily. Once dismounted, a cyclist becomes a pedestrian (at least in most countries). So, on a hw=primary, the meaning of bicycle=no becomes clear upon inspecting the corresponding foot tag. If there is no such tag or foot=yes, walking the bike is fine. In case of foot=no, it is not.

**Comment by Phyks** [comment]
> That is exactly the point. hw=pedestrian should have the same rendering as hw=pedestrian+bicycle=no (or the rare hw=pedestrian+bicycle=dismount) but it does not.

Indeed, due to https://github.com/cyclosm/cyclosm-cartocss-style/blob/master/roads.mss#L2895-L2897.

I made a PR to try to improve this. I'm waiting for @Florimondable feedback on this one.

**Comment by Florimondable** [comment]
As said by the first sentence of the page it's a proposal
«Default is a proposal (not yet fully drafted, no RFC, no voting yet) for a possible, upcoming default value system.»

I hope, but may be that's not  the case, that pedestrian roads are rendered in CyclOSM in such a way that we understand it's not a perfect road for cyclist and can be quickly seen as pedestrian.

**Comment by Phyks** [comment]
I dug back on this and through the country-specific restrictions. My main question was basically "can the default proposal be seen as a majority vote across country-specific restrictions?". Basically, are a majority of countries having a bias towards `bicycle=dismount` or `bicycle=yes`, so that default rendering could accomodate a majority of people worldwide.

It is definitely not the case, and it seems roughly half of the countries are a `bicycle=yes` and half of them are a `bicycle=dismount|no`. :/

The question of having country-specific values built into CyclOSM is happening more and more often (maxspeeds, driving side, etc.). We should probably investigate how this could be achieved (practically, in terms of rendering and code, not in terms of parsing OSM tags / wiki), but this is a separate issue.

Concerning this specific question of default restriction for `highway=pedestrian`, **I don't have strong opinion. I would be slightly more enclined to bias towards the `bicycle=dismount` which would be more conservative (in compliance with default proposal, no legal risk to indicate you cannot cycle when you actually can).**
